### ISSUE TITLE
text in fences should not be linted

### ISSUE THREAD

**Issue by Granitosaurus** [igorshubovych/markdownlint-cli#92]
Text in fences (\`\`\`) is being linted eventhough it's not markdown.

e.g.

```
This is some text:

```python
def function():<space>
    pass
```

will throw a MD009 error when it shouldn't

```

**Comment by DavidAnson** [comment]
That's intended - there's usually no more reason to use trailing space inside a fence than outside.

The rendering is the same either way:

```python
def function(): 
    pass
```

```python
def function():
    pass
```

Does python require the trailing space in this scenario?

**Comment by Granitosaurus** [comment]
@DavidAnson I'd disagree with you here. Markdown linter should only lint _markdown_. Linting in code-fences create unnecessary conflicts and friction and while trailing spaces might not be required they are most certainly allowed and functional in python.   
I'd also imagine there would be some sort language or format that does require trailing spaces, just saying that no text every should have or even tollerate trailing spaces is a bit extreme.

**Comment by DavidAnson** [comment]
Any scenarios, files, or parts of files that require trailing spaces can disable MD009/no-trailing-spaces either for the project, the file, or the relevant part of the file. This rule (and others) is a suggestion, not a requirement.

The topic has come up a couple of times before and opting out of the rule seemed a fine option. It would be possible to add a `code_blocks` parameter to this rule to allow trailing spaces in a code block (similar to MD010/no-hard-tabs), but I'd want to understand the motivation first. Trailing spaces are not meaningful in the vast majority of scenarios and almost never visible in the editor or when rendered to HTML.

Just like `git` warns of trailing spaces in a file, so does `markdownlint`. I think about `markdownlint` as working on Markdown **files** (which is why YAML frontmatter is supported and inline HTML recognized) and so don't draw a strong distinction between content inside/outside a code block.

**Comment by DavidAnson** [comment]
This should be addressed by https://github.com/DavidAnson/markdownlint/issues/292
